server/src/routes/upload/parseBeatmap.ts

```diff
@@ -87,6 +91,11 @@ export const parseBeatmap: (
     throw ERR_BEATMAP_AUDIO_INVALID
   }
 
+  const songDuration = await getDurationInSeconds(
+    `${infoJSON._songAuthorName}-${infoJSON._songName}`,
```

Comment:
See my other comment about temporary directories. Consider passing in `infoJSON._songFilename`, and the map's `hash` (you'll need to move the function call after the hash calculation) instead.

Reply:
Please see above comment.  I opted to focus on uniqueness, but still passed in the `hash` (even though it's not necessary), but it might be useful for identifying the file for future debugging